web/src/pages/playground.tsx

```diff
@@ -1,387 +1,62 @@
-  updateSettings: (updater: (settings: LLMSettings) => LLMSettings) => void;
-};
-
-const ChatPrompts = ({
-  settings,
-  query,
-  history,
-  updateSettings,
-}: ChatPromptParams) => {
-  const updatePrompt =
-    (...path: string[]) =>
-    (event: ChangeEvent) => {
-      const newPrompts = {
-        ...updateIn(
-          settings.prompts || {},
-          path,
-          (event.target as HTMLInputElement).value
-        ),
-      };
-      updateSettings((settings) => ({ ...settings, prompts: newPrompts }));
-    };
-
-  return (
-    <div className="chat-prompts mx-5 w-[400px] flex-none border-2 p-5 outline-black">
-      <details open>
-        <summary>Source prompt</summary>
-        <TextareaAutosize
-          className="border-gray w-full border px-1"
-          value={settings?.prompts?.context}
-          onChange={updatePrompt("context")}
-        />
-        <div>(This is where sources will be injected)</div>
-      </details>
-      {history.length > 0 && (
-        <details open>
-          <summary>History prompt</summary>
-          <TextareaAutosize
-            className="border-gray w-full border px-1"
-            value={settings?.prompts?.history}
-            onChange={updatePrompt("history")}
-          />
-          <details>
-            <summary>History</summary>
-            {history
-              .slice(Math.max(0, history.length - (settings.maxHistory || 0)))
-              .map((entry, i) => (
-                <div className="history-entry" key={i}>
-                  {entry.content}
-                </div>
-              ))}
-          </details>
-        </details>
-      )}
-      <details open>
-        <summary>Question prompt</summary>
-        <TextareaAutosize
-          className="border-gray w-full border px-1"
-          value={settings?.prompts?.question}
-          onChange={updatePrompt("question")}
-        />
-        <TextareaAutosize
-          className="border-gray w-full border px-1"
-          value={settings?.prompts?.modes[settings.mode || "default"]}
-          onChange={updatePrompt("modes", settings.mode || "default")}
-        />
-      </details>
-      <div>Q: {query}</div>
-    </div>
-  );
-};
+import {
+  ChatSettings,
+  ChatPrompts,
+  updateIn,
+  makeSettings,
+} from "../components/settings";
 
 const Playground: NextPage = () => {
   const [sessionId, setSessionId] = useState("");
-  const [settings, updateSettings] = useState<LLMSettings>(DEFAULT_SETTINGS);
+  const [settings, updateSettings] = useState<LLMSettings>(makeSettings({}));
 
   const [query, setQuery] = useState<string>("");
   const [history, setHistory] = useState<Entry[]>([]);
 
+  const router = useRouter();
+
   const setMode = (mode: [Mode, boolean]) => {
     if (mode[1]) {
       localStorage.setItem("chat_mode", mode[0]);
       updateSettings((settings) => ({ ...settings, mode: mode[0] }));
     }
   };
 
+  const changeSetting = (path: string[], value: any) => {
+    router.replace(
```

Comment:
this will add any changes in the various widgets to the URL, so it'll be a lot easier to save/share parameters